pkg/serverless/logs/logs.go

```diff
@@ -27,27 +28,18 @@ type Tags struct {
 	Tags []string
 }
 
-// ExecutionContext represents the execution context
-type ExecutionContext struct {
-	ARN                string
-	LastRequestID      string
-	ColdstartRequestID string
-	LastLogRequestID   string
-	Coldstart          bool
-	StartTime          time.Time
-}
-
 // LambdaLogsCollector is the route to which the AWS environment is sending the logs
 // for the extension to collect them.
 type LambdaLogsCollector struct {
 	LogChannel             chan *logConfig.ChannelMessage
 	Demux                  aggregator.Demultiplexer
 	ExtraTags              *Tags
-	ExecutionContext       *ExecutionContext
 	LogsEnabled            bool
 	EnhancedMetricsEnabled bool
 	// HandleRuntimeDone is the function to be called when a platform.runtimeDone log message is received
-	HandleRuntimeDone func()
+	HandleRuntimeDone                  func()
+	GetExecutionContext                func() executioncontext.ExecutionContext
+	UpdateExecutionContextFromStartLog func(string, time.Time)
```

Comment:
I would rather not expose this function from here, but more as a method in `executioncontext.ExecutionContext`
Something like : 
`func (ec *ExecutionContext) UpdateExecutionContextFromStartLog`
What do you think?